### ISSUE TITLE
arrow-parens option for refs

### ISSUE THREAD

**Issue by pvinis** [eslint/eslint#9571]
**Please describe what the rule should do:**
in the same vain as `requireForBlockBody`, i would like to suggest `notRequiredForRef`.
basically, i like having `arrow-parens` on `'always'`, since it makes adding args easy, but for refs, its always the same, always `ref={r => this.bla = r}` and i think its nicer this way, with no parens.

**What category of rule is this? (place an "X" next to just one item)**

[X] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
<div ref={(r) => this.someDiv = r}>
  <p>hi</p>
</div>
```

**Why should this rule be included in ESLint (instead of a plugin)?**
i guess it makes sense to add this as an additional option.

**Comment by not-an-aardvark** [comment]
Thanks for the proposal. Personally I don't think I'm in favor of this proposal, because the behavior of `ref` is a React-specific detail which doesn't apply to JSX. However, let's see what the rest of the team thinks.

**Comment by pvinis** [comment]
thank you for the reply. ah you mean it could go to the react plugin?

**Comment by mysticatea** [comment]
> you mean it could go to the react plugin?

I think so.

I agree with @not-an-aardvark.
ESLint core rules should be enough general. If it depends on specific library/framework, it should be a rule of a plugin.

I will close this issue because above.
